test/e2e/rule_test.go

```diff
@@ -556,6 +560,141 @@ func TestRule(t *testing.T) {
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(querier))
+
+	r, err := e2ethanos.NewRuler(s.SharedDir(), "1", rulesSubDir, []alert.AlertmanagerConfig{
+		{
+			EndpointsConfig: http_util.EndpointsConfig{
+				FileSDConfigs: []http_util.FileSDConfig{
+					{
+						// FileSD which will be used to register discover dynamically am1.
+						Files:           []string{filepath.Join(e2e.ContainerSharedDir, amTargetsSubDir, "*.yaml")},
+						RefreshInterval: model.Duration(time.Second),
+					},
+				},
+				StaticAddresses: []string{
+					am.NetworkHTTPEndpoint(),
+				},
+				Scheme: "http",
+			},
+			Timeout:    model.Duration(time.Second),
+			APIVersion: alert.APIv1,
+		},
+	}, []query.Config{
+		{
+			EndpointsConfig: http_util.EndpointsConfig{
+				// We test Statically Addressed queries in other tests. Focus on FileSD here.
+				FileSDConfigs: []http_util.FileSDConfig{
+					{
+						// FileSD which will be used to register discover dynamically q.
+						Files:           []string{filepath.Join(e2e.ContainerSharedDir, queryTargetsSubDir, "*.yaml")},
+						RefreshInterval: model.Duration(time.Second),
+					},
+				},
+				Scheme: "http",
+			},
+		},
+	}, true,  remotewrite.Config{
+		Name: "ruler-rw-receivers",
+		RemoteStore: &config.RemoteWriteConfig{
+			URL: &commoncfg.URL{URL: rwURL},
+			Name: "thanos-receiver",
+		},
+	})
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(r))
+
+	writeTargets(t, filepath.Join(s.SharedDir(), queryTargetsSubDir, "targets.yaml"), querier.NetworkHTTPEndpoint())
+	writeTargets(t, filepath.Join(s.SharedDir(), amTargetsSubDir, "targets.yaml"), am.NetworkHTTPEndpoint())
+
+	t.Run("inject samples into receiver to reset its StoreAPI MinTime", func(t *testing.T) {
+		// inject data into receiver to reset its minTime (so it doesn't get filtered out by store)
+		// the sample is injected through a prometheus instance that remote_writes samples into the receiver node
+		prom, _, err := e2ethanos.NewPrometheus(s.SharedDir(), "1", defaultPromConfig("prom", 0, e2ethanos.RemoteWriteEndpoint(receiver.NetworkEndpoint(8081)), ""), e2ethanos.DefaultPrometheusImage())
+		testutil.Ok(t, err)
+		testutil.Ok(t, s.StartAndWaitReady(prom))
+
+		queryAndAssertSeries(t, ctx, querier.HTTPEndpoint(), queryUpWithoutInstance, promclient.QueryOptions{
+			Deduplicate: false,
+		}, []model.Metric{
+			{
+				"job":"myself",
+				"prometheus": "prom",
+				"receive": "1",
+				"replica": "0",
+				"tenant_id": "default-tenant",
+			},
+		})
+	})
+
+	t.Run("query can contact from receiver", func(t *testing.T) {
+		testAbsentQuery := "absent(nonexistent{job='thanos-receive'})"
+		queryAndAssertSeries(t, ctx, querier.HTTPEndpoint(), testAbsentQuery, promclient.QueryOptions{
+			Deduplicate: false,
+		}, []model.Metric{
+			{
+				"job": "thanos-receive",
+			},
+		})
+	})
+
+	t.Run("can fetch remote-written samples from receiver", func(t *testing.T) {
```

Comment:
I think this should work and let's just without data generation wait for that! (: